src/compiler.rs

```diff
@@ -492,6 +525,48 @@ impl<CM: CellManager, SSB: StepSelectorBuilder> Compiler<CM, SSB> {
         }
     }
 
+    fn place_fixed_assignments<F: Field>(
+        &self,
+        unit: &CompilationUnit<F>,
+        assignments: FixedAssignment<F>,
+    ) -> Assignment<F> {
+        let mut result = Assignment::default();
+        let step_height = unit.placement.first_step_height();
+        let empty = vec![F::ZERO; unit.num_rows];
+
+        for (queriable, assignments) in assignments {
+            let placement = match queriable {
+                Queriable::Fixed(fixed, rot) => {
+                    if rot != 0 {
+                        panic!("cannot do fixed assignation of rotated queriable");
+                    }
+                    unit.placement.get_fixed_placement(&fixed)
+                }
+                Queriable::Halo2FixedQuery(signal, rot) => SignalPlacement::new(
+                    unit.find_halo2_fixed(signal).expect("column not found"),
+                    rot,
+                ),
+                _ => panic!("only can do fixed assignment to fixed signal"),
+            };
+
+            let mut column_values = if let Some(column_values) = result.get(&placement.column) {
+                column_values.clone()
+            } else {
+                empty.clone()
+            };
+
+            let mut offset = placement.rotation as usize;
+            for value in assignments {
+                column_values[offset] = value;
+                offset += step_height as usize;
+            }
```

Comment:
Looks like we limit the total number of fixed rows to the number of step instances. This isn't necessarily true, especially when we have large lookup tables that contain fixed columns. Any reason for this design?

It also looks like fixed column assignments start from step instance zero and increment by 1 for each assignment. I'm not sure what's the purpose of tying step instance index to fixed assignments? Also if a step's height is greater than one, we only assign one row for the fixed column, which can be a waste of space.